sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -1792,3 +1793,138 @@ def _df_set_column_unicode_key_impl(self, key, value):
 
     ty_checker = TypeChecker('Method _set_column().')
     ty_checker.raise_exc(key, 'str', 'key')
+
+
+def sdc_pandas_dataframe_reset_index_codegen(drop, all_params, columns):
+    """
+    Example of generated implementation:
+        def _df_reset_index_impl(self, level=None, drop=False, inplace=False, col_level=0, col_fill=""):
+          length = len(get_dataframe_data(self, 0))
+          res_index = numpy.arange(length)
+          result_0 = get_dataframe_data(self, 0)
+          return pandas.DataFrame({"index": res_index, "A": result_0}, index=numpy.arange(len(self)))
+    """
+    result_name = []
+    all_params_str = ', '.join(all_params)
+    func_lines = [f'def _df_reset_index_impl({all_params_str}):']
+    df = all_params[0]
+    if not drop.literal_value:
+        old_index = 'old_index'
+        func_lines += [f'  {old_index} = {df}.index']
+        result_name.append((old_index, 'index'))
+    for i, c in enumerate(columns):
+        result_c = 'result_c'
+        func_lines += [
+            f'  {result_c} = get_dataframe_data({df}, {i})'
+        ]
+        result_name.append((result_c, c))
```

Comment:
So you will add the same (last) column into the DF many times, check with:
df = pd.DataFrame({'A': [1.0, 2.0, np.nan, 1.0], 'B': np.arange(4.0)}, index=[5, 8, 4, 6])